src/Features/CSharp/Portable/EditAndContinue/SyntaxUtilities.cs

```diff
@@ -83,7 +83,7 @@ public static void AssertIsBody(SyntaxNode syntax, bool allowLambda)
             if (LambdaUtilities.IsLambdaBody(syntax))
             {
                 Debug.Assert(allowLambda);
-                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax);
+                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax || syntax is ArrowExpressionClauseSyntax);
```

Comment:
I think that would be contrary to a lot of the assumptions in other code. In the same way the block is the body of a regular method, not the statement list, the `ArrowExpression` is the body of an expression-bodied local function.